### PR TITLE
[Feature][SeaTunnel Engine IMap Storage] Imap storage supports kafka compact topic in cluster mode

### PR DESCRIPTION
**Pull request by sunxiaojian**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by sunxiaojian** [comment]
@EricJoy2048 PTAL

**Comment by EricJoy2048** [comment]
@liugddx  PTAL.

**Review by liugddx — COMMENTED** [review]
Why are the test cases disabled? We need to add some test cases that can be used to ensure that this feature can be used.

**Comment by sunxiaojian** [comment]
> Why are the test cases disabled? We need to add some test cases that can be used to ensure that this feature can be used.

The test is connected to my local environment

**Comment by liugddx** [comment]
> > Why are the test cases disabled? We need to add some test cases that can be used to ensure that this feature can be used.
> 
> The test is connected to my local environment

Is there a way to enable testing?

**Comment by sunxiaojian** [comment]
> Is there a way to enable testing?

Can be processed using kafka container

**Comment by EricJoy2048** [comment]
> 

Yes, I think you can enable the test case by use kafka container.

**Comment by sunxiaojian** [comment]
> > 
> 
> Yes, I think you can enable the test case by use kafka container.

done

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/pom.xml [hunk 1]
```diff
@@ -92,6 +99,19 @@
...
+
+        <dependency>
+            <groupId>org.testcontainers</groupId>
+            <artifactId>kafka</artifactId>
+            <version>${testcontainer.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
```
Comment (liugddx) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/pom.xml:123]: Do we need this dependency?

File: pom.xml [hunk 2]
```diff
@@ -146,6 +146,7 @@
         <json-smart.version>2.4.7</json-smart.version>
         <hadoop-aws.version>3.1.4</hadoop-aws.version>
         <netty-buffer.version>4.1.60.Final</netty-buffer.version>
+        <kafka.version>3.4.1</kafka.version>
```
Comment (liugddx) [inline pom.xml]: We support those versions of kafka?

File: seatunnel-engine/seatunnel-engine-storage/imap-storage-plugins/imap-storage-kafka/pom.xml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            <groupId>org.testcontainers</groupId>
+            <artifactId>kafka</artifactId>
+            <version>${testcontainer.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.13.2</version>
+            <scope>test</scope>
```
Comment (liugddx) [inline seatunnel-engine/seatunnel-engine-storage/imap-storage-plugins/imap-storage-kafka/pom.xml:68]: Same as above.

Comment (sunxiaojian) [reply seatunnel-engine/seatunnel-engine-storage/imap-storage-plugins/imap-storage-kafka/pom.xml:68]: > Same as above.

for testcontainer

File: pom.xml (hunk 2 above)
Comment (sunxiaojian) [reply pom.xml]: > We support those versions of kafka?

Backward compatibility.

File: seatunnel-engine/seatunnel-engine-storage/imap-storage-plugins/imap-storage-kafka/pom.xml (hunk 3 above)
Comment (liugddx) [reply seatunnel-engine/seatunnel-engine-storage/imap-storage-plugins/imap-storage-kafka/pom.xml:68]: https://github.com/apache/seatunnel/blob/6060b13f9f9a223fd0a85c33d98e5f86df3aa254/pom.xml#L530C9-L530C9

**Comment by liugddx** [comment]
I have a question, why use kafka as storage?

**Comment by sunxiaojian** [comment]
> I have a question, why use kafka as storage?

Kafka compact topic:

1. Support message merging with the same key, automatically achieve file compression, effectively improve restart loading speed
2. Supports tombstone messages. During the data compression and merging process, tombstone messages can be cleared to reduce storage space
3. Support for unstructured storage, Conforming to Zeta engine storage scenarios （Can refer to the design of Kafka connect）
4. Open source, widely used

File: seatunnel-engine/seatunnel-engine-storage/imap-storage-plugins/imap-storage-kafka/pom.xml (hunk 3 above)
Comment (liugddx) [inline seatunnel-engine/seatunnel-engine-storage/imap-storage-plugins/imap-storage-kafka/pom.xml:68]: @EricJoy2048 @Hisoka-X PTAL

**Review by liugddx — DISMISSED** [review]
In general LGTM.

**Comment by ic4y** [comment]
The latest dev branch has addressed e2e test issues. Please consider merging it.

**Comment by sunxiaojian** [comment]
> The latest dev branch has addressed e2e test issues. Please consider merging it.

@ic4y There are still errors

**Comment by liugddx** [comment]
Waiting for https://github.com/apache/seatunnel/pull/5208

**Comment by sunxiaojian** [comment]
@liugddx @ic4y PTAL

**Pull request closed without merging**